CMakeLists.txt

```
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.28)
+
+project(flasher-stub CXX)
+
+option(TARGET_CHIP "Target ESP chip" "OFF")
+
+if("${TARGET_CHIP}" STREQUAL "OFF")
+    message(FATAL_ERROR "Please set target chip via -DTARGET_CHIP.")
+endif()
+
+# CXX standard set based on the GCC version of the toolchain for ESP8266 (oldest one)
+set(CUSTOM_FLAGS --std=gnu++17 -Wall -Wextra -Werror -Wshadow -Wundef -Wconversion -Os -fno-common -nostdlib -fno-builtin -Wl,-static -g -ffunction-sections -Wl,--gc-sections)
+set(EXTRA_RISCV_FLAGS -march=rv32imc -mabi=ilp32 -msmall-data-limit=0)
+set(EXTRA_XTENSA_FLAGS -mtext-section-literals -mlongcalls)
+
+set(ESP8266_FLAGS "-DESP8266" ${EXTRA_XTENSA_FLAGS})
+set(XTENSA_FLAGS "-DXTENSA" ${EXTRA_XTENSA_FLAGS})
+set(RISCV_FLAGS "-DRISCV" ${EXTRA_RISCV_FLAGS})
+
+message(STATUS "Building for ${TARGET_CHIP}")
+add_executable(${TARGET_CHIP})
+
+set(ESP32_XTENSA_CHIPS esp32 esp32-s2 esp32-s3)
```

If there is no specific reason, can we remove the dash? Afaik, our naming convention is `esp32s2` or `ESP32-S2` 
```suggestion
set(ESP32_XTENSA_CHIPS esp32 esp32s2 esp32s3)
```

So that it will be consistent with the chip directory names in the esp-stub-lib

---

I planned to do this refactoring later but moved it up and did it here.